src/ssh/cert.rs

```diff
@@ -409,9 +411,8 @@ impl Certificate {
     /// Take the certificate settings and generate a valid signature using the provided signer function
     pub fn sign(mut self, signer: impl FnOnce(&[u8]) -> Option<Vec<u8>>) -> Result<Self> {
         let mut writer = super::Writer::new();
-        let kt_name = format!("{}-cert-v01@openssh.com", self.key.key_type.name);
         // Write the cert type
-        writer.write_string(kt_name.as_str());
+        writer.write_string(self.key.key_type.name);
```

Comment:
The change in `builder` and `sign` cancel each other out? If that's correct I'm tempted to leave it the way it is because it's more tested and doesn't require mutation.